IntelliTectAnalyzer/IntelliTectAnalyzer/Analyzers/NamingFieldPascalUnderscore.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace IntelliTectAnalyzer.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NamingFieldPascalUnderscore : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "FieldNaming";
+        private const string Title = "IntelliTect naming violation";
+        private const string MessageFormat = "Fields should be named _PascalCase";
+        private const string Description = "Adjust naming";
+        private const string Category = "Naming";
+
+        private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Field);
+        }
+
+        private static void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var namedTypeSymbol = context.Symbol;
+
+            if (namedTypeSymbol.Name.StartsWith("_"))
+            {
+                if (namedTypeSymbol.Name.Length > 1 && char.IsUpper(namedTypeSymbol.Name.Skip(1).First()))
```

Comment:
You can reduce nesting by moving all of the if conditions into a single if statement with `&&`